src/modules/flow/http-client/http-client.c

```diff
@@ -714,6 +714,276 @@ float_post_process(struct sol_flow_node *node, void *data, uint16_t port, uint16
 }
 
 /*
+ * --------------------------------- rgb node  -----------------------------
+ */
+
+static int
+rgb_process_token(struct sol_flow_node *node, struct sol_json_token *key,
+    struct sol_json_token *value)
+{
+    struct sol_rgb rgb = { 0 };
+    enum sol_json_loop_reason reason;
+    struct sol_json_scanner sub_scanner;
+    struct sol_json_token sub_key, sub_value, token;
+
+    sol_json_scanner_init(&sub_scanner, value->start,
+        value->end - value->start);
+    SOL_JSON_SCANNER_OBJECT_LOOP (&sub_scanner, &token, &sub_key,
+        &sub_value, reason) {
+        if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&sub_key, "red")) {
+            if (sol_json_token_get_uint32(&sub_value, &rgb.red) < 0)
+                return -EINVAL;
+        } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&sub_key, "green")) {
+            if (sol_json_token_get_uint32(&sub_value, &rgb.green) < 0)
+                return -EINVAL;
+        } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&sub_key, "blue")) {
+            if (sol_json_token_get_uint32(&sub_value, &rgb.blue) < 0)
+                return -EINVAL;
+        } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&sub_key, "red_max")) {
+            if (sol_json_token_get_uint32(&sub_value, &rgb.red_max) < 0)
+                return -EINVAL;
+        } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&sub_key, "green_max")) {
+            if (sol_json_token_get_uint32(&sub_value, &rgb.green_max) < 0)
+                return -EINVAL;
+        } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&sub_key, "blue_max")) {
```

Comment:
hum.... I think we should declare `rgb` with better defaults for `red_max` and the likes, otherwise if we do not produce them in the JSON the resulting rgb will be correct. Maybe check if components are `<=` their maximum values before emitting the packet.